ngraph_bridge/ngraph_encapsulate_impl.cc

```diff
@@ -588,6 +588,55 @@ void NGraphEncapsulateImpl::DumpNgFunction(
   StringToFile(file_name, m_serialized_ng_function_map[ng_exec]);
 }
 
+void NGraphEncapsulateImpl::NGraphEncapsulateImpl::ClearExecMaps() {
+  m_ng_exec_input_cache_map.clear();
+  m_ng_exec_output_cache_map.clear();
+  m_ng_exec_map.clear();
+  m_serialized_ng_function_map.clear();
+  m_executable_pipelined_tensors_map.clear();
+}
+
+Status NGraphEncapsulateImpl::GetPipelineIdxAndTensors(
+    const std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
+    std::tuple<int, PipelinedTensorVector, PipelinedTensorVector>& tpl) {
+  if (GetExecCanCreateTensor()) {
```

Comment:
Do we need this if?

Reply:
Yes.

If we call `UpdatePipelinedTensorCache` or `GetTensorsFromPipeline` without the `if`, they will throw errors.

Reply:
The calling function have the same check
https://github.com/tensorflow/ngraph-bridge/blob/master/ngraph_bridge/ngraph_encapsulate_op.cc#L328.

Maybe its better that way so the user knows what it is calling